libdebug/debugger/internal_debugger.py

```diff
@@ -838,50 +859,64 @@ def _craft_gdb_migration_command(self: InternalDebugger) -> list[str]:
 
         return gdb_command + bp_args
 
-    def _open_gdb_in_new_process(self: InternalDebugger) -> None:
+    def _open_gdb_in_new_process(self: InternalDebugger, args: list[str]) -> None:
         """Opens GDB in a new process following the configuration in libcontext.terminal."""
-        args = self._craft_gdb_migration_command()
-
         initial_pid = Popen(libcontext.terminal + args).pid
 
         os.waitpid(initial_pid, 0)
 
-        liblog.debugger("Waiting for GDB process to terminate...")
+        def wait_for_termination() -> None:
+            liblog.debugger("Waiting for GDB process to terminate...")
 
-        for proc in psutil.process_iter():
-            try:
-                cmdline = proc.cmdline()
-            except psutil.ZombieProcess:
-                # This is a zombie process, which psutil tracks but we cannot interact with
-                continue
+            for proc in psutil.process_iter():
+                try:
+                    cmdline = proc.cmdline()
+                except psutil.ZombieProcess:
+                    # This is a zombie process, which psutil tracks but we cannot interact with
+                    continue
 
-            if args == cmdline:
-                gdb_process = proc
-                break
-        else:
-            raise RuntimeError("GDB process not found.")
+                if args == cmdline:
+                    gdb_process = proc
+                    break
+            else:
+                raise RuntimeError("GDB process not found.")
+
+            while gdb_process.is_running() and gdb_process.status() != psutil.STATUS_ZOMBIE:
+                # As the GDB process is in a different group, we do not have the authority to wait on it
+                # So we must keep polling it until it is no longer running
+                pass
 
-        while gdb_process.is_running() and gdb_process.status() != psutil.STATUS_ZOMBIE:
-            # As the GDB process is in a different group, we do not have the authority to wait on it
-            # So we must keep polling it until it is no longer running
-            pass
+        return wait_for_termination
 
-    def _open_gdb_in_shell(self: InternalDebugger) -> None:
+    def _open_gdb_in_shell(self: InternalDebugger, args: list[str]) -> None:
         """Open GDB in the current shell."""
         gdb_pid = os.fork()
+
         if gdb_pid == 0:  # This is the child process.
-            args = self._craft_gdb_migration_command()
             os.execv("/bin/gdb", args)
-        else:  # This is the parent process.
-            # Parent ignores SIGINT, so only GDB (child) receives it
-            signal.signal(signal.SIGINT, signal.SIG_IGN)
+            raise RuntimeError("Failed to execute GDB.")
```

Comment:
are there cases in which the os.execv returns and not raise and exception?

Reply:
I don't think so, but it's there just in case...

Reply:
Unfortunately Python doesn't support `__builtin_unreachable()`